### PR TITLE
Fix dependencies for 0.32

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
Hey @FabianHofmann and @fneum, it looks like some dependencies weren't quite right in #18.  This PR is aimed at fixing them and also some other clean-up:
* fixing the license name
* adding doc/dev URLs
* updating the `script` line to match grayskull

I hope that's good with you all.  I also added myself as a maintainer so I hope you're good with that, too.

**Comment by xylar** [comment]
I was looking things over because of this forum question:
https://conda.discourse.group/t/xarray-error-message-using-linopy/495
Not sure if this update will fix that issue but I'm hoping it might.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,22 +11,23 @@ source:
...
   run:
-    - python >=3.8
+    - python >=3.9
     - numpy
     - scipy
     - bottleneck
     - toolz
     - numexpr
-    - xarray >=2022.9.0
+    - xarray >=2023.1.0
```
Comment (FabianHofmann) [inline recipe/meta.yaml:30]:
```suggestion
    - xarray >=2023.9.0
```

**Review by FabianHofmann — COMMENTED** [review]
Great, thanks for the contribution! We can even add a tighter xarray restriction.

File: recipe/meta.yaml (hunk 1 above)
Comment (xylar) [reply recipe/meta.yaml:30]: Yes, I saw that that had been updated after I made the PR.  Let's make that change.

**Comment by xylar** [comment]
@FabianHofmann, this is now ready to merge, I believe.

**Comment by FabianHofmann** [comment]
Great! Thanks and welcome on board!

**Pull request merged**